### PR TITLE
#297 Parse and offer to import public key received by email

### LINKED ISSUE #297: Parse and offer to import public key received by email
Follow Android functionality. When the message had a public key, you will see one or more `publicKey` MsgBlocks

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented function to parse and offer to import public key received by email

close #297 // if this PR closes an issue

https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/c90782c7-3b76-446a-8c64-dff844a93e87

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCryptUI/Nodes/PublicKeyDetailNode.swift [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    public var onImportKey: (() -> Void)?
+
+    let input: Input
+
+    public init(input: Input) {
+        self.input = input
+        super.init()
+        backgroundColor = UIColor(hex: "FAFAFA")
```
Comment (sosnovsky) [inline FlowCryptUI/Nodes/PublicKeyDetailNode.swift:136]: Need to change color for dark mode:

Screenshot 2023-09-13 at 15 47 00

File: FlowCryptUI/Cell Nodes/CellNode.swift [hunk 2]
```diff
@@ -9,6 +9,23 @@
 import AsyncDisplayKit
 
 open class CellNode: ASCellNode {
+    var leftBorder: ASDisplayNode?
+
+    func addLeftBorder(width: CGFloat, color: UIColor?) {
```
Comment (sosnovsky) [inline FlowCryptUI/Cell Nodes/CellNode.swift:14]: This left border is a nice addition 👍 But let's show it only for encrypted messages, while plain messages won't have this border, so it'll be easier for users to differentiate between secure encrypted messages and plain messages.

**Review by sosnovsky — COMMENTED** [review]
Works great 👍

File: FlowCryptUI/Nodes/PublicKeyDetailNode.swift (hunk 1 above)
Comment (ioanmo226) [reply FlowCryptUI/Nodes/PublicKeyDetailNode.swift:136]: Nice catch 👍  Let me fix

File: FlowCryptUI/Cell Nodes/CellNode.swift (hunk 2 above)
Comment (ioanmo226) [reply FlowCryptUI/Cell Nodes/CellNode.swift:14]: Ok

Comment (ioanmo226) [reply FlowCryptUI/Cell Nodes/CellNode.swift:14]: I have another idea.
I think better to add another border color for unecrypted message? something like lightgray or something like that?

Comment (sosnovsky) [reply FlowCryptUI/Cell Nodes/CellNode.swift:14]: Sounds good 👍

File: FlowCryptUI/Cell Nodes/MessageTextSubjectNode.swift [hunk 3]
```diff
@@ -53,7 +55,7 @@ public final class MessageTextSubjectNode: CellNode {
         if let quote = input.quote {
             setupTextNode(quoteNode, text: quote, accessibilityIdentifier: "aid-message-\(input.index)-quote")
         }
-        addLeftBorder(width: .threadLeftBorderWidth, color: .main)
+        addLeftBorder(width: .threadLeftBorderWidth, color: input.isEncrypted ? .main : UIColor(hex: "777777"))
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCryptUI/Cell Nodes/MessageTextSubjectNode.swift:58]: Ok

**Review by sosnovsky — APPROVED** [review]
Thanks!

**Comment by sosnovsky** [comment]
I'll merge this one after https://github.com/FlowCrypt/flowcrypt-ios/pull/2347, so it'll go to the next release

**Pull request merged**